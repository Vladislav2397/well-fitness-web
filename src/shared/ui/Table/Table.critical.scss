.table {
    &__item {
        display: flex;
        flex-direction: row;

        & + & {
            border-top: 1px solid #ccc;
        }
    }
}
