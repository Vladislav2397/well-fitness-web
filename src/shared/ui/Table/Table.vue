<template lang="pug">

.b-table
    .__item(
        v-for="item in list"
    )
        slot(
            :item="item"
        )

</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

export type TableProps = {
    //
}

@Component
export default class Table extends Vue {
    @Prop() readonly list!: unknown[]
}
</script>

<!-- <style lang="scss" src="./Table.critical.scss" /> -->
<!-- <style lang="scss" src="./Table.main.scss" /> -->
