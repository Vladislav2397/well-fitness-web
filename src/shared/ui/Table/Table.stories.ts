import { Story } from '@storybook/vue'
import { action } from '@storybook/addon-actions'

import Table, { type TableProps } from './Table.vue'
import { defineControls } from '@/shared/lib/stories'

export default {
    title: 'undefined-path/Table',
    component: Table,
    argTypes: defineControls({
        //
    }),
}

const Template: Story<TableProps> = (args, { argTypes }) => ({
    props: Object.keys(argTypes),
    components: { Table },
    methods: {
        handle: action('clicked'),
    },
    template: `<Table v-bind="$props" @click="handle" />`,
})

export const Default = Template.bind({})
Default.args = {
    //
}
