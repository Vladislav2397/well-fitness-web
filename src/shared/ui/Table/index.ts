import Table, { type TableProps } from './Table.vue'

export { Table, TableProps }
