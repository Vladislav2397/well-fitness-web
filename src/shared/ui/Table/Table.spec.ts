import { shallowMount } from '@vue/test-utils'
import Table /*, { type TableProps }*/ from './Table.vue'

describe('test Table', () => {
    test('snapshot', () => {
        const wrapper = shallowMount(Table, {
            propsData: {
                //
            } /* as TableProps*/,
        })
        expect(wrapper).toMatchSnapshot()
    })
})
