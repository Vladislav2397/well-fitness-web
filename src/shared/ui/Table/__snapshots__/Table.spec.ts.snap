// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Table snapshot 1`] = `<div class="b-table"></div>`;
